pkg/clusternetwork/config.go

```diff
@@ -77,3 +107,19 @@ func getNetworkType(oclient configclient.Interface) (string, error) {
 	}
 	return networkCR.Spec.NetworkType, nil
 }
+
+// getServiceNetworkCIDR gets the serviceCIDR using cluster config
+// this is required for cni configuration
+func getServiceNetworkCIDR(oclient configclient.Interface) (string, error) {
+	// Get the cluster network object so that we can find the service network
+	networkCR, err := oclient.ConfigV1().Networks().Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Wrap(err, "error getting cluster network object")
+	}
+	serviceCIDR := networkCR.Spec.ServiceNetwork[0]
+	_, _, err = net.ParseCIDR(serviceCIDR)
+	if err != nil || serviceCIDR == "" {
+		return "", errors.Wrapf(err, "error receiving valid value for cluster service CIDR")
+	}
```

Comment:
Can't you use `validateCIDR` here too? I am guessing you will have a cyclic dependency which is why you have not done so. But that just points at our package layout not being ideal.

Reply:
yes avoided to not have cyclic dependency